### PR TITLE
[ONNX] Update back the mutated buffers into the original PyTorch model

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113686
* #112272
* #113495

Currently, when the user_model: torch.export.ExportedProgram is exported
to ONNX and its implementation mutates buffers, the mutated buffer is
returned as an extra output in the ONNX graph while the original model
still have the outdated buffer value.

If we do a numerical comparison between the pytorch model and the ONNX
model, this mismatch is detected, failing the numerical equivalence
check

This PR updates the pytorch model with the new buffer value after the
ONNX model is executed, making both models numerically equivalent.

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
@BowenBao PTAL

File: torch/onnx/_internal/exporter.py [hunk 1]
```diff
@@ -660,7 +663,48 @@ def __call__(self, *args: Any, options=None, **kwargs: Any) -> Any:
...
+        if isinstance(self._model_torch, torch.export.ExportedProgram):
+            self.update_mutated_buffers_onto_torch_model(
+                self._model_torch.module(), onnxruntime_outputs
+            )
+        return onnxruntime_outputs
+
+    def update_mutated_buffers_onto_torch_model(
+        self,
+        torch_model: torch.nn.Module,
+        onnxruntime_outputs: Sequence[
```
Comment (BowenBao) [inline torch/onnx/_internal/exporter.py:677]:
```suggestion
        onnx_outputs: Sequence[
```
nit: doesn't have to be runtime specific?

**Review by BowenBao — COMMENTED** [review]
What is the motivation? Feels like `ExportOutput` aka `ONNXProgram` now is expanding in size in a considerably fast rate.

**Comment by thiagocrepaldi** [comment]
Closing in favor of an updated https://github.com/pytorch/pytorch/pull/112272

**Pull request closed without merging**